src/pages/Main/components/hooks/useFetchAllOfPosts.ts

```diff
@@ -0,0 +1,23 @@
+import { keepPreviousData, useInfiniteQuery } from '@tanstack/react-query';
+import { AxiosError } from 'axios';
+import { getFullPostList } from '@/apis/getPosts';
+
+interface UseFetchAllOfPostsParams {
+  limit: number;
+}
+
+const useFetchAllOfPosts = ({ limit }: UseFetchAllOfPostsParams) =>
+  useInfiniteQuery({
+    queryKey: ['posts', 'all'],
+    queryFn: ({ pageParam = 1 }) =>
+      getFullPostList({ limit, offset: pageParam }),
+    retry: (failureCount, error: AxiosError) =>
+      error.status === 502 && failureCount < 3,
+    getNextPageParam: (lastPage, _, lastPageParmas) =>
+      lastPage.length < limit ? undefined : lastPageParmas + limit,
+    initialPageParam: 0,
+    select: (data) => data.pages.flatMap((page) => page),
+    placeholderData: keepPreviousData,
+  });
+
+export default useFetchAllOfPosts;
```

Comment:
이부분도 의견 같이 남겨드립니다. ! 

수혁님이 이부분을 사용하는 곳이 `PostList.tsx` 에서 사용하는데 props로 `useInfinityQuery` 의 모든 내용을 전달하다보니 props의 타입정의도 복잡해지는것 같아요 !

두가지 수정을 거치면 가독성이 좋아질거라고 예상하는데
1번 

<img width="491" alt="image" src="https://github.com/prgrms-fe-devcourse/FEDC5_NodakNodak_Noah/assets/102784200/be787a3e-6a2b-46c3-88d2-48e9b06dd8ed">

이 커스텀훅을 AllPostList.tsx에서 굳이 호출하지 않아도 될 것 같아요 !

<img width="542" alt="image" src="https://github.com/prgrms-fe-devcourse/FEDC5_NodakNodak_Noah/assets/102784200/559a3510-a4ed-4003-bf44-07d7969bfcf6">

여기서 직접 커스텀훅을 호출하면 어떨까 싶습니다 ! 군데군데 제안사항이 갈려있어서 뭔가 여기에 한번에 남겨놓는게 좋을거라고 생각했어요 ! 

2번 
위에 `select`는 data의 변환값인걸로 예상되네요 ! 그렇게 하는것보다
```suggestion

return {
 postList : data.pages.map((page) => page).flat(),
 fetchNextPage,
 hasNextPage,
 isFetchingNextPage
}

```
이렇게 리턴해주면 바로 사용할 수 있을 것 같아요 !

Reply:
searchedPost 와 Post 데이터가 달라서,  PostList props 의 타입을 정의하기가 어려웠던것 같아요!
searchedPost 는 useQueryResult (pagination api 부재를 이유로..) , Post 는 useInfiniteQueryResult를 내려줍니다!

호민님이 지적 해주셔서, 열심히 가독성 좋게 수정을 해봤는데요,
```ts
// ChannelPostList.tsx

const ChannelPostList = () => {
  const channelId = useParams<{ channelId: string }>().channelId;
  const response = useFetchChannelPosts({ channelId, limit: 9 });

  return <PostList {...response} />;
};
```
```ts
// PostList.tsx

type PostListProps = Partial<UseInfiniteQueryResult<PostCardProps[]>> &
  UseQueryResult<PostCardProps[]>;

const PostList = ({
  data,
  isFetching,
  isPending,
  fetchNextPage,
  hasNextPage,
  isFetchingNextPage,
}: PostListProps) => {
  if (isPending) {
    return <MainPageSpinner />;
  }

  return ( /...
```
이런식으로 가독성을 높일수있습니다.! 이런식은 혹시 어떻게 생각하시나요?!

useFetch 훅으로 묶어서구현하지않고, 굳이 이런식으로 구현한이유는,, postList 에 관련된 useFetch 훅이 총 3개 (search api, channelId post list api, all post list api) 있는데, PostList.tsx 에서 3개 모두 호출하는것이 의외로, 복잡도가 높더라고요! 그래서 책임이 분리되면 좋겠다고 생각해서, link로 바로 이어지는 allPostList.tsx, searchPostList.tsx, channelPostList.tsx 로 각각 useFetch 훅을 사용한것이고요!

근데 사실 그렇게 차이는 크지않은것 같아요 특히 pagination 을 버리고 infinity scroll 로 바꾸니 오히려, 너무 분리를 많이한것같기도하네요?!

Reply:
음 일단 useFetch훅을 각각 분리해준건 너무나 좋은 분리방법인 것 같아요 ! 
근데 다른 분리내용은 회의떄 이야기하는게 빠르게 논의가 끝날 것 같네요 :) 시간잡아서 한번 따로 이야기 해보면 어떨까요